components/gantry/oneaxis/oneaxis.go

```diff
@@ -172,88 +172,54 @@ func (g *oneAxis) home(ctx context.Context, np int) error {
 	defer done()
 	// Mapping one limit switch motor0->limsw0, motor1 ->limsw1, motor 2 -> limsw2
 	// Mapping two limit switch motor0->limSw0,limSw1; motor1->limSw2,limSw3; motor2->limSw4,limSw5
+
+	var positionA, positionB float64
+	var err error
 	switch np {
+	// An axis with an encoder will encode the zero position, and add the second position limit
+	// based on the steps per length
+	case 0:
+		revPerLength := g.lengthMm / g.mmPerRevolution
+		positionA, err = g.motor.Position(ctx, nil)
+		if err != nil {
+			return err
+		}
+		positionB = positionA + revPerLength
 	// An axis with one limit switch will go till it hits the limit switch, encode that position as the
 	// zero position of the oneaxis, and adds a second position limit based on the steps per length.
 	case 1:
-		if err := g.homeOneLimSwitch(ctx); err != nil {
+		positionA, err = g.testLimit(ctx, true)
+		if err != nil {
+			return err
+		}
+		revPerLength := g.lengthMm / g.mmPerRevolution
+		positionB = positionA + revPerLength
+
+		x := g.gantryToMotorPosition(0.2 * g.lengthMm)
+		if err = g.motor.GoTo(ctx, g.rpm, x, nil); err != nil {
 			return err
 		}
 	// An axis with two limit switches will go till it hits the first limit switch, encode that position as the
 	// zero position of the oneaxis, then go till it hits the second limit switch, then encode that position as the
 	// at-length position of the oneaxis.
 	case 2:
-		if err := g.homeTwoLimSwitch(ctx); err != nil {
+		positionA, err = g.testLimit(ctx, true)
+		if err != nil {
 			return err
 		}
-	// An axis with an encoder will encode the
-	case 0:
-		if err := g.homeEncoder(ctx); err != nil {
+		positionB, err = g.testLimit(ctx, false)
+		if err != nil {
+			return err
+		}
+		x := g.gantryToMotorPosition(0.8 * g.lengthMm)
+		if err = g.motor.GoTo(ctx, g.rpm, x, nil); err != nil {
```

Comment:
let's also check if the difference between the positions is 0 or if either position A or B is `math.Nan` and error if so. Maybe it's better to initialize the positions as math.Nan to be explicit about this now that I think about it.

Reply:
I was getting lint errors initializing values as `math.NaN` and comparing values to `math.Nan` so I removed that and it's now just checking if the difference between A and B is 0